src/main/java/com/iexec/core/task/TaskRepository.java

```diff
@@ -44,6 +44,42 @@
     @Query("{ 'currentStatus': {$in: ?0} }")
     List<Task> findByCurrentStatus(List<TaskStatus> statuses, Sort sort);
 
+    @Query("{ 'currentStatus': {$in: ?0} , 'tag': {$ne: ?1} }")
+    List<Task> findByCurrentStatusInAndTagNot(List<TaskStatus> statuses, String excludedTag, Sort sort);
+
+    /**
+     * Shortcut for {@link TaskRepository#findFirstByCurrentStatusInAndTagNotAndChainTaskIdNotIn}.
+     * Retrieves the prioritized task matching with given criteria:
+     * <ul>
+     *     <li>Task is in one of given {@code statuses};</li>
+     *     <li>Task has not given {@code excludedTag}
+     *          - this is mainly used to exclude TEE tasks;
+     *     </li>
+     *     <li>Chain task ID is not one of the given {@code excludedChainTaskIds};</li>
+     *     <li>Tasks are prioritized according to the {@code sort} parameter.</li>
+     * </ul>
+     *
+     * @param statuses             The task status should be one of this list.
+     * @param excludedTag          The task tag should not be this tag
+     *                             - use {@literal null} if no tag should be excluded.
+     * @param excludedChainTaskIds The chain task ID should not be one of this list.
+     * @param sort                 How to prioritize tasks.
+     * @return The first task matching with the criteria, according to the {@code sort} parameter.
+     */
+    default Optional<Task> findPrioritizedTask(List<TaskStatus> statuses,
```

Comment:
Why not in `TaskService`? For me the repository should contain the least amount of logic and only define the interface with repositories.

Reply:
The only point of `findPrioritizedTask` is to avoid using `findFirstByCurrentStatusInAndTagNotAndChainTaskIdNotIn` outside of this repo. This method's name is way too long so I supposed it would be better to add an alias directly into the repo.

Reply:
If we move `findPrioritizedTask` to the service, the long-named method will only be called once. Other services should call `TaskService`.

Reply:
Alright, see https://github.com/iExecBlockchainComputing/iexec-core/pull/496 :)